processor/otel/logs_test.go

```diff
@@ -0,0 +1,150 @@
+			assert.Empty(t, batch)
+			return nil
+		}
+
+		consumer := otel.Consumer{Processor: processor}
+		logs := pdata.NewLogs()
+		assert.NoError(t, consumer.ConsumeLogs(context.Background(), logs))
+	})
+
+	commonEvent := model.APMEvent{
+		Processor: model.LogProcessor,
+		Agent: model.Agent{
+			Name:    "otlp/go",
+			Version: "unknown",
+		},
+		Service: model.Service{
+			Name:     "unknown",
+			Language: model.Language{Name: "go"},
+		},
+		Message: "a random log message",
+		Event: model.Event{
+			Severity: int64(pdata.SeverityNumberINFO),
+			Action:   "doOperation()",
+		},
+		Log:   model.Log{Level: "Info"},
+		Span:  &model.Span{ID: "0200000000000000"},
+		Trace: model.Trace{ID: "01000000000000000000000000000000"},
+		Labels: model.Labels{
+			"key": model.LabelValue{Value: "value"},
+		},
+		NumericLabels: model.NumericLabels{
+			"numeric_key": model.NumericLabelValue{Value: 1234},
+		},
+	}
+	test := func(name string, body interface{}, expectedMessage string) {
+		t.Run(name, func(t *testing.T) {
+			logs := newLogs(body)
+
+			var processed model.Batch
+			var processor model.ProcessBatchFunc = func(_ context.Context, batch *model.Batch) error {
+				if processed != nil {
+					panic("already processes batch")
+				}
+				processed = *batch
+				assert.NotNil(t, processed[0].Timestamp)
+				processed[0].Timestamp = time.Time{}
+				return nil
+			}
+			consumer := otel.Consumer{Processor: processor}
+			assert.NoError(t, consumer.ConsumeLogs(context.Background(), logs))
+
+			expected := commonEvent
+			expected.Message = expectedMessage
+			assert.Equal(t, model.Batch{expected}, processed)
+		})
+	}
+	test("string_body", "a random log message", "a random log message")
+	test("int_body", 1234, "1234")
+	test("float_body", 1234.1234, "1234.1234")
+	test("bool_body", true, "true")
+	// TODO(marclop): How to test map body
+}
+
+func newLogs(body interface{}) pdata.Logs {
+	logs := pdata.NewLogs()
+	resourceLogs := logs.ResourceLogs().AppendEmpty()
+	logs.ResourceLogs().At(0).Resource().Attributes().InitFromMap(map[string]pdata.AttributeValue{
+		semconv.AttributeTelemetrySDKLanguage: pdata.NewAttributeValueString("go"),
+	})
+	instrumentationLogs := resourceLogs.InstrumentationLibraryLogs().AppendEmpty()
+	otelLog := instrumentationLogs.Logs().AppendEmpty()
+	otelLog.SetTraceID(pdata.NewTraceID([16]byte{1}))
+	otelLog.SetSpanID(pdata.NewSpanID([8]byte{2}))
+	otelLog.SetName("doOperation()")
+	otelLog.SetSeverityNumber(pdata.SeverityNumberINFO)
+	otelLog.SetSeverityText("Info")
+	otelLog.SetTimestamp(pdata.NewTimestampFromTime(time.Now()))
+	otelLog.Attributes().InitFromMap(map[string]pdata.AttributeValue{
+		"key":         pdata.NewAttributeValueString("value"),
+		"numeric_key": pdata.NewAttributeValueDouble(1234),
+	})
+
+	switch b := body.(type) {
+	case string:
+		otelLog.Body().SetStringVal(b)
+	case int:
+		otelLog.Body().SetIntVal(int64(b))
+	case float64:
+		otelLog.Body().SetDoubleVal(float64(b))
```

Comment:
```suggestion
		otelLog.Body().SetDoubleVal(b)
```

it's already float64